OrbitService/OrbitService.cpp

```diff
@@ -1,27 +1,165 @@
+  process_list_.Refresh();
+  process_list_.UpdateCpuTimes();
+  std::string process_data = SerializeObjectHumanReadable(process_list_);
+  tcp_server_->Send(Msg_RemoteProcessList, process_data.data(),
+                    process_data.size());
+}
+
+void OrbitService::SetupServerCallbacks() {
+  tcp_server_->AddMainThreadCallback(
+      Msg_RemoteProcessRequest, [this](const Message& msg) {
+        auto pid =
+            static_cast<uint32_t>(msg.m_Header.m_GenericHeader.m_Address);
+        SendProcess(pid);
+      });
+
+  tcp_server_->AddMainThreadCallback(
+      Msg_RemoteSelectedFunctionsMap,
+      [this](const Message& msg) { SetSelectedFunctions(msg); });
 
-  ConnectionManager::Get().InitAsService();
+  tcp_server_->AddMainThreadCallback(
+      Msg_StartCapture, [this](const Message& msg) {
+        auto pid =
+            static_cast<uint32_t>(msg.m_Header.m_GenericHeader.m_Address);
+        StartCapture(pid);
+      });
 
-  transaction_service_ = std::make_unique<TransactionService>(GTcpServer.get());
+  tcp_server_->AddMainThreadCallback(Msg_StopCapture,
+                                     [this](const Message&) { StopCapture(); });
+}
+
+void OrbitService::SendProcess(uint32_t pid) {
+  LOG("Sending info on process %d", pid);
+  std::shared_ptr<Process> process = process_list_.GetProcess(pid);
+  if (process != nullptr) {
+    // TODO: Remove this: pid should be part of every message
+    //  and all the messages should to be as stateless as possible.
+    process->ListModules();
+    process->EnumerateThreads();
+    std::string process_data = SerializeObjectHumanReadable(*process);
+    tcp_server_->Send(Msg_RemoteProcess, process_data.data(),
+                      process_data.size());
+  }
+}
+
+void OrbitService::SetSelectedFunctions(const Message& message) {
+  LOG("Received selected functions");
+  DeserializeObjectBinary(message.GetData(), message.GetSize(),
+                          selected_functions_);
+}
+
+void OrbitService::StartCapture(uint32_t pid) {
+  LOG("Starting capture");
+  tracing_handler_.Start(pid, selected_functions_);
+  tracing_buffer_thread_ = std::thread{[this] { TracingBufferThread(); }};
+}
+
+void OrbitService::StopCapture() {
+  LOG("Stopping capture");
+  tracing_handler_.Stop();
+  if (tracing_buffer_thread_.joinable()) {
+    tracing_buffer_thread_.join();
+  }
+}
+
+void OrbitService::SetupTransactionServices() {
+  transaction_service_ = std::make_unique<TransactionService>(tcp_server_);
   symbols_service_ = std::make_unique<SymbolsService>(
-      &ConnectionManager::Get().GetProcessList(), transaction_service_.get());
+      &process_list_, transaction_service_.get());
   process_memory_service_ =
       std::make_unique<ProcessMemoryService>(transaction_service_.get());
 }
 
-void OrbitService::Run(std::atomic<bool>* exit_requested) {
-  while (!(*exit_requested)) {
-    GTcpServer->ProcessMainThreadCallbacks();
-    Sleep(16);
+void OrbitService::TracingBufferThread() {
```

Comment:
Most of this code should probably be moved to OrbitAsioService - since this is asio specific code.

Reply:
I added a commit for that, please have a look.